src/Mailgun/Connection/RestClient.php

```diff
@@ -147,6 +147,9 @@ public function post($endpointUrl, array $postData = [], $files = [])
                     $postDataMultipart[] = [
                         'name' => sprintf('%s[%d]', $key, $index++),
                         'contents' => $subValue,
+                        'headers' => [
+                            'Content-Disposition' => sprintf('form-data; name="%s"', $key),
```

Comment:
Just using `$key` here will not be good. If one adds multiple resources named "foobar" it will be overwritten because the server will only see one resource named "foobar". That is why we need the array syntax.